code/game/machinery/big_manipulator/big_manipulator.dm

```diff
@@ -0,0 +1,909 @@
+      point_overlays += point_appearance
+
+  for(var/i = 1 in 1 to length(pickup_points))
+    var/datum/interaction_point/point = dropoff_points[i]
+    var/turf/target_turf = point.interaction_turf
+    if(target_turf)
+      var/mutable_appearance/point_appearance = mutable_appearance('icons/effects/interaction_points.dmi', "dropoff_[i]", ABOVE_NORMAL_TURF_LAYER, src, src.plane)
+      var/turf/manip_turf = get_turf(src)
+      point_appearance.pixel_x = (target_turf.x - manip_turf.x) * 32
+      point_appearance.pixel_y = (target_turf.y - manip_turf.y) * 32
+      point_overlays += point_appearance
+
+  main_hud.overlays += point_overlays
+  hud_points += main_hud
+  set_hud_image_active(BIG_MANIP_HUD)
+
+/// Attempts to find the closest open turf to the manipulator
+/obj/machinery/big_manipulator/proc/find_suitable_turf()
+	var/turf/center = get_turf(src)
+
+	var/list/directions = GLOB.alldirs
+	for(var/dir in directions)
+		var/turf/checked_turf = get_step(center, dir)
+		if(checked_turf && !isclosedturf(checked_turf))
+			return checked_turf
+
+	// didn't find any :boowomp:
+	return null
+
+/// Attempts to create a new interaction point and assign it to the correct list.
+/// Arguments: `new_turf` (turf), `new_filters` (list), `new_filters_status` (boolean),
+/// `new_interaction_mode` (use a define), `transfer_type` (use a define).
+/obj/machinery/big_manipulator/proc/create_new_interaction_point(mob/user, turf/new_turf, list/new_filters, new_filters_status, new_interaction_mode, transfer_type)
+	if(!new_turf)
+		new_turf = find_suitable_turf()
+		if(!new_turf)
+			balloon_alert(user, "no suitable turfs found!")
+			return FALSE
+
+	if(transfer_type == TRANSFER_TYPE_PICKUP && length(pickup_points) + 1 > interaction_point_limit)
+		balloon_alert(user, "pickup point limit reached!")
+		return FALSE
+	if(transfer_type == TRANSFER_TYPE_DROPOFF && length(dropoff_points) + 1 > interaction_point_limit)
+		balloon_alert(user, "dropoff point limit reached!")
+		return FALSE
+
+	var/datum/interaction_point/new_interaction_point = new(new_turf, new_filters, new_filters_status, new_interaction_mode)
+
+	if(QDELETED(new_interaction_point)) // if something STILL somehow went wrong
+		return FALSE
+
+	switch(transfer_type) // assigning to the correct list
+		if(TRANSFER_TYPE_PICKUP)
+			pickup_points += new_interaction_point
+		if(TRANSFER_TYPE_DROPOFF)
+			dropoff_points += new_interaction_point
+
+	if(obj_flags & EMAGGED)
+		new_interaction_point.type_filters += /mob/living
+
+	if(is_operational)
+		update_hud()
+
+	return new_interaction_point
+
+/// Allow each point to interact with mobs when the manipulator is emagged.
+/obj/machinery/big_manipulator/proc/update_all_points_on_emag_act()
+	for(var/datum/interaction_point/pickup_point in pickup_points)
+		pickup_point.type_filters += /mob/living
+	for(var/datum/interaction_point/dropoff_point in dropoff_points)
+		dropoff_point.type_filters += /mob/living
```

Comment:
this proc existing is odd (the single use + the fact that is named referencing that it is only used once), why not just put this in the emag act call?